src/config/job-config/jobconfig.interface.ts

```diff
@@ -0,0 +1,85 @@
+import { CreateJobDto } from "../../jobs/dto/create-job.dto";
+import { StatusUpdateJobDto } from "../../jobs/dto/status-update-job.dto";
+import { JobsAuth } from "../../jobs/types/jobs-auth.enum";
+import { JobClass } from "../../jobs/schemas/job.schema";
+
+// Nest Token for CREATE job actions
+export const CREATE_JOB_ACTION_FACTORIES = Symbol(
+  "CREATE_JOB_ACTION_FACTORIES",
+);
+// Nest Token for UPDATE job actions
+export const UPDATE_JOB_ACTION_FACTORIES = Symbol(
+  "UPDATE_JOB_ACTION_FACTORIES",
+);
+
+export interface JobConfigListOptions {
+  configVersion?: string;
+  jobs: JobConfigOptions[];
+}
+
+/**
+ * Encapsulates all responses to a particular job type (eg "archive")
+ */
+export interface JobConfig {
+  jobType: string;
+  configVersion: string;
+  create: JobOperation<CreateJobDto>;
+  statusUpdate: JobOperation<StatusUpdateJobDto>;
+}
+export interface JobConfigOptions {
```

Comment:
`JobConfigOptions` represents the POD object parsed from yaml. This is then converted to the real `JobConfig` instance. They have similar structure except that the `JobConfig` has `JobAction` instances at its leaves, while `JobConfigOptions` has the configuration strings as leaves.

Reply:
It's possible this could be done directly in the yaml parser as a custom deserializer, but this follows the patterns used in Nest's documentation for custom providers.